integrations/DCMFloodlight/browser.js

```diff
@@ -103,155 +124,73 @@ class DCMFloodlight {
     countingMethod = countingMethod.trim().toLowerCase().replace(/\s+/g, "_");
 
     // find conversion event
-    // some() stops execution if at least one condition is passed and returns bool
     // knowing cat (activityTag), type (groupTag), (counter or sales), customVariable from config
-    event = event.trim().toLowerCase();
-    const conversionEventFound = this.conversionEvents.some(
-      (conversionEvent) => {
-        if (
-          conversionEvent &&
-          conversionEvent.eventName &&
-          conversionEvent.eventName.trim().toLowerCase() === event
-        ) {
-          if (
-            isNotEmpty(conversionEvent.floodlightActivityTag) &&
-            isNotEmpty(conversionEvent.floodlightGroupTag)
-          ) {
-            this.activityTag = conversionEvent.floodlightActivityTag.trim();
-            this.groupTag = conversionEvent.floodlightGroupTag.trim();
-          }
-          salesTag = conversionEvent.salesTag;
-          customFloodlightVariable = conversionEvent.customVariables || [];
-          return true;
-        }
-        return false;
-      }
+    const conversionEvent = this.conversionEvents.find(
+      (cnEvent) =>
+        cnEvent &&
+        cnEvent.eventName &&
+        cnEvent.eventName.trim().toLowerCase() === event.toLowerCase()
     );
 
-    if (!conversionEventFound) {
+    if (!conversionEvent) {
       logger.error("[DCM Floodlight]:: Conversion event not found");
       return;
     }
 
+    if (
+      conversionEvent.floodlightActivityTag &&
+      conversionEvent.floodlightGroupTag
+    ) {
+      this.activityTag = conversionEvent.floodlightActivityTag.trim();
+      this.groupTag = conversionEvent.floodlightGroupTag.trim();
+    }
+
+    const { salesTag } = conversionEvent;
+
+    if (!isValidCountingMethod(salesTag, countingMethod)) {
+      logger.error(
+        `[DCM Floodlight] ${
+          salesTag ? "Sales" : "Counter"
+        } Tag:: invalid counting method`
+      );
+      return;
+    }
+
+    customFloodlightVariable = conversionEvent.customVariables || [];
     customFloodlightVariable = transformCustomVariable(
       customFloodlightVariable,
       message
     );
 
-    // Ref - https://support.google.com/campaignmanager/answer/7554821?hl=en#zippy=%2Ccustom-fields
-    // we can pass custom variables to DCM and any values passed in it will override its default value
-    // Total 7 properties - ord, num, dc_lat, tag_for_child_directed_treatment, tfua, npa, match_id
```

Comment:
Please provide proper comments when this implementation is moved as in how many properties are there and rest along with reference

Reply:
Updated the custom variables (params) references in `buildGtagTrackPayload` and `buildIframeTrackPayload` . Please check